cmd/baton-cloudflare/main.go

```diff
@@ -6,40 +6,66 @@ import (
 	"os"
 
 	"github.com/conductorone/baton-cloudflare/pkg/connector"
-	"github.com/conductorone/baton-sdk/pkg/cli"
+	configSchema "github.com/conductorone/baton-sdk/pkg/config"
 	"github.com/conductorone/baton-sdk/pkg/connectorbuilder"
+	"github.com/conductorone/baton-sdk/pkg/field"
 	"github.com/conductorone/baton-sdk/pkg/types"
 	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/spf13/viper"
 	"go.uber.org/zap"
 )
 
-var version = "dev"
+var (
+	apiKeyField         = field.StringField(apiKey, field.WithDescription("The api key for the Cloudflare account."))
+	apiTokenField       = field.StringField(apiToken, field.WithDescription("The api token for the Cloudflare account."))
+	accountIdField      = field.StringField(accountId, field.WithRequired(true), field.WithDescription("The account id for the Cloudflare account."))
+	emailIdField        = field.StringField(emailId, field.WithDescription("The email id for the Cloudflare account."))
+	configurationFields = []field.SchemaField{apiKeyField, apiTokenField, accountIdField, emailIdField}
+	fieldRelationships  = []field.SchemaFieldRelationship{
+		field.FieldsAtLeastOneUsed(apiTokenField, apiKeyField),
+		field.FieldsDependentOn(
+			[]field.SchemaField{apiKeyField},
+			[]field.SchemaField{emailIdField},
+		),
+	}
+)
+
+const (
+	version       = "dev"
+	connectorName = "baton-cloudflare"
+	apiKey        = "api-key"
+	apiToken      = "api-token"
+	accountId     = "account-id"
+	emailId       = "email-id"
+)
 
 func main() {
 	ctx := context.Background()
-	cfg := &config{}
-	cmd, err := cli.NewCmd(ctx, "baton-cloudflare", cfg, validateConfig, getConnector)
+	_, cmd, err := configSchema.DefineConfiguration(ctx,
+		connectorName,
+		getConnector,
+		field.NewConfiguration(configurationFields, fieldRelationships...),
+	)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
 
 	cmd.Version = version
-	cmdFlags(cmd)
 	err = cmd.Execute()
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err.Error())
 		os.Exit(1)
 	}
 }
 
-func getConnector(ctx context.Context, cfg *config) (types.ConnectorServer, error) {
+func getConnector(ctx context.Context, cfg *viper.Viper) (types.ConnectorServer, error) {
 	l := ctxzap.Extract(ctx)
 	config := connector.Config{
-		AccountId: cfg.AccountId,
-		ApiToken:  cfg.ApiToken,
-		EmailId:   cfg.EmailId,
-		ApiKey:    cfg.ApiKey,
+		AccountId: cfg.GetString(accountId),
```

Comment:
```suggestion
		AccountId: cfg.GetString(accountIdField.FieldName),
```
You can use `.FieldName` for all of these, allowing you to put the actual names in the calls to `field.StringField(...)` and delete the consts on lines 33-40.